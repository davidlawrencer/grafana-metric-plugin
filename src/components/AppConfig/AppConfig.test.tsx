import React from 'react';
import {render} from '@testing-library/react';
import {PluginType} from '@grafana/data';
import {Config, Props} from './AppConfig';

describe('Components/AppConfig', () => {
    let props: Props;

    beforeEach(() => {
        jest.resetAllMocks();

        props = {
            plugin: {
                meta: {
                    id: 'sample-app',
                    name: 'Sample App',
                    type: PluginType.app,
                    enabled: false,
                    jsonData: {},
                },
            },
            query: {},
        } as unknown as Props;
    });

    test('renders without an error"', () => {
        const screen = render(<Config plugin={props.plugin} query={props.query}/>);
        expect(screen.container).toMatchSnapshot();
    });

    test('renders an "Enable" button if the plugin is disabled', () => {
        const plugin = {meta: {...props.plugin.meta, enabled: false}};

        // @ts-ignore - We don't need to provide `addConfigPage()` and `setChannelSupport()` for these tests
        const screen = render(<Config plugin={plugin} query={props.query}/>);

        expect(screen.queryByText('Enable')).toBeInTheDocument();
        expect(screen.queryByText('Disable')).not.toBeInTheDocument();
    });

    test('renders a "Disable" button if the plugin is enabled', () => {
        const plugin = {meta: {...props.plugin.meta, enabled: true}};

        // @ts-ignore - We don't need to provide `addConfigPage()` and `setChannelSupport()` for these tests
        const screen = render(<Config plugin={plugin} query={props.query}/>);

        expect(screen.queryByText('Disable')).toBeInTheDocument();
        expect(screen.queryByText('Enable')).not.toBeInTheDocument();
    });
});
