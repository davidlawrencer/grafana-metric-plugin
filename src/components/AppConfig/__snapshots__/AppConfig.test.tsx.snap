// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/AppConfig renders without an error" 1`] = `
<div>
  <h2>
    Embrace Mobile
  </h2>
  <p>
    The Embrace Mobile App is a plugin for Grafana that gives teams an easy way to visualize Embrace dashboards using Embrace Metrics API data powered by Prometheus.
  </p>
  <p>
    Get out-of-the-box Embrace dashboards in minutes with our simple configuration. Enter your Embrace API Token below and click to Enable the Application.
  </p>
  <form>
    <label>
      Embrace API Token: Â 
      <input
        size="50"
        type="text"
        value=""
      />
    </label>
  </form>
  <div
    class="gf-form gf-form-button-row"
  >
    <button
      class="css-1sara2j-button"
      disabled=""
      type="button"
    >
      <span
        class="css-1mhnkuh"
      >
        Enable
      </span>
    </button>
  </div>
</div>
`;
